src/Defines.h

```diff
@@ -425,6 +431,24 @@ extern eDamageType StringToDamageType(const AString & a_DamageString);
 If a_Inverse is true, the opposite direction is used instead. */
 void AddFaceDirection(int & a_BlockX, int & a_BlockY, int & a_BlockZ, eBlockFace a_BlockFace, bool a_bInverse = false);
 
+/** a_CurrentPosiiton - the block of whom we would like to get the block next to
+a_BlockFace - the face (direction) in which we would like to get the block
+*/
+inline Vector3i GetBlockNextTo(Vector3i a_CurrentPosition, eBlockFace a_BlockFace)
```

Comment:
Is this not the same as the function above: `AddFaceDirection`?

Reply:
Similar, but more convenient and using vectors. I would go as far to say that AddFaceDirection should be removed entirely in the future (when we completely switch over to Vector3). AddFaceDirection could be argued to be slightly faster, but if you have to do the inverse at some point I think that's moot.

Reply:
Yes, but why call it `GetBlockNextTo` and have less features when it could just be:
```cpp
inline Vector3i AddFaceDirection(Vector3i a_BlockPos, eBlockFace a_BlockFace, bool a_bInverse = false)
{
	AddFaceDirection(a_BlockPos.x, a_BlockPos.y, a_BlockPos.z, a_BlockFace, a_bInverse);
	return a_BlockPos;
}
```

Reply:
Makes sense, seems like a sensible change

Reply:
I've extracted this to a separate PR, because I'd like to use that in my own PR soon.
#4666 .